javascript/extraNetworks.js

```diff
@@ -224,4 +220,141 @@ function updateAllPrivatePreviewsAndMonitorChanges() {
+        }
+
+        const metaDataButtonNode = document.createElement('div');
+        metaDataButtonNode.className = 'metadata-button';
+        metaDataButtonNode.title = "Show metadata";
+        metaDataButtonNode.setAttribute('onclick', `extraNetworksRequestMetadata(event, "${model_type}", "${item.name}")`);
+
+        const actionsNode = document.createElement('div');
+        actionsNode.className = 'actions';
+
+        const additionalNode = document.createElement('div');
+        additionalNode.className = "additional";
+
+        const ulNode = document.createElement('ul');
+        const aNode = document.createElement('a');
+        aNode.title = "replace preview image with currently selected in gallery";
+        aNode.setAttribute('onclick', `return saveCardPreview(event, "${tabname}", "${model_type}/${item.name}.png")`);
+        aNode.target = "_blank";
+        aNode.innerHTML = "set private preview";
+
+        ulNode.appendChild(aNode);
+
+        const searchTermNode = document.createElement('span');
+        searchTermNode.className = "search_term";
+        searchTermNode.style.display = "none";
+        searchTermNode.innerHTML = item.search_term;
+
+        const nameNode = document.createElement('span');
+        nameNode.className = "name";
+        nameNode.innerHTML = item.name;
+
+        const descriptionNode = document.createElement('span');
+        descriptionNode.className = "description";
+
+        additionalNode.appendChild(ulNode);
+        additionalNode.appendChild(searchTermNode);
+
+        actionsNode.appendChild(additionalNode);
+        actionsNode.appendChild(nameNode);
+        actionsNode.appendChild(descriptionNode);
+
+        cardNode.appendChild(metaDataButtonNode);
+        cardNode.appendChild(actionsNode);
+        cardsParentNode.insertBefore(cardNode, uploadBtnNode);
+    })
+}
+
+function updatePage(tabname, model_type, page_type) {
+    let currentPage = 1;
+    let totalCount;
+    
+    const currentPageTabsId = `${tabname}_${model_type}_current_page`;
+    const currentTotalCountId = `${tabname}_${model_type}_total_count`;
+
+    currentPage = currentPageForTabs.get(currentPageTabsId);
+    totalCount = totalCountForTabs.get(currentTotalCountId);
+
+    if (currentPage === 1 && page_type === 'previous') {
+        return
+     }
+     if (currentPage * pageSize >= totalCount && page_type === 'next') {
+        return
+     }
+    const page = page_type === 'previous' ? currentPage - 1 : currentPage + 1;
+    fetchPageDataAndUpdateList({ tabname, model_type, page });
+}
+
+function setPageSize() {
+    const contentWidth = document.body.clientWidth - 84;
+    pageSize = Math.floor(contentWidth / 238) * 2;
+}
+
+async function refreshModelList({tabname}) {
+    const tabs = gradioApp().querySelector('#'+tabname+'_extra_tabs > div')
+    const model_type = tabs.querySelector('.selected').innerHTML.toLowerCase().replace(/^\s+|\s+$/g,"").replace(" ", "_");

```

Comment:
Same issue here, is there a better way to set this in the python code to make it consistent with backend.